### ISSUE TITLE
reorganizing the filters file

### ISSUE THREAD

**Issue by lea-hagen** [BEAST-Fitting/beast#160]
Currently, the filters are in beast/libs/filters.hd5, which the user downloads from box along with some other files.  It would be nice to have the filters stored in such a way that if someone adds a filter, they could do a pull request to have it available to everyone.

After some discussion at the hack day, we came up with a few options to make this feasible:

1. Put the filters.hd5 file in the repository: This would be mildly annoying, since it's a binary file.
2. Change the format to ASDF before adding to repo: would require re-writing the code that accesses it
3. Store the filters as ascii files and add to repo: also would require re-writing BEAST code
4. Store filters as as ascii files, and then have some code that runs (perhaps at the beginning of making the model grids) to create the filters.hd5 file

I would personally lean towards 4.  Anyone else have additional ideas?

**Comment by karllark** [comment]
The 4th option is probably the best.  Would need to get the ascii files for all the existing filters in the filters.hd5 file for this to work.  

I would suggest that the code to generate the filters.hd5 file only be run if the filters.hd5 file is not already present.  May not be needed to recreate it every time the beast is run.  The generation code could even check if there are any new filters and add only the new filters to the filters.hd5 file.

**Comment by lea-hagen** [comment]
I'm guessing that it'll be reasonably straightforward to create ascii files using the current filters.hd5 file.

I agree on that second part - I can have it do a check on both the file existing and which filters are there.

**Comment by mfouesneau** [comment]
If you integrate `pyphot` you can do very simply number 3.

**Comment by lea-hagen** [comment]
@mfouesneau random question - do you recall where the filter curves came from?  (we think that you assembled filters.hd5?)

**Comment by mfouesneau** [comment]
Yes I did. Various databased from facilities (e.g. CFHT), the SVO database, and IRAF library.

**Comment by karllark** [comment]
In related work, I've started to look at synphot and stsynphot.  Looks like this might be a good way to go.  At least for HST stuff, it looks like we wouldn't have to do any work and someone else would maintain things.  Still investigating though.

https://github.com/spacetelescope/synphot_refactor
https://github.com/spacetelescope/stsynphot_refactor

**Comment by mfouesneau** [comment]
FYI, https://github.com/mfouesneau/pyphot now also deals with astropy units and its preliminary version was the one integrated into BEAST at that time.
